custom_components/tuya_local/devices/ultenic_t10_vacuum.yaml

```diff
@@ -0,0 +1,121 @@
+name: Ultenic T10 Robot vacuum
+products:
+  - id: skn6zhhv0uxrylrk
+    name: Ultenic T10
+primary_entity:
+  entity: vacuum
+  dps:
+    - id: 1
+      type: boolean
+      name: power
+      optional: true
+    - id: 2
+      type: boolean
+      name: activate  
+      optional: true
+    - id: 13
+      type: boolean
+      name: locate
+      optional: true
+    - id: 3
+      type: string
+      name: command
+      mapping:
+        - dps_val: standby
+          value: standby
+        - dps_val: random
+          value: random
+        - dps_val: smart
+          value: smart
+        - dps_value: wall_follow
+          value: wall_follow
+        - dps_val: mop
+          value: mop
+        - dps_val: spiral
+          value: clean_spot
+        - dps_val: left_spiral
+          value: left_spiral
+        - dps_val: right_spiral
+          value: right_spiral
+        - dps_val: left_bow
+          value: left_bow
+        - dps_val: right_bow
+          value: right_bow
+        - dps_val: partial_bow
+          value: partial_bow
+        - dps_val: chargego
+          value: return_to_base
+    - id: 5
+      type: string
+      name: status
+      mapping:
+        - dps_val: standby
+          value: standby
+        - dps_val: smart_clean
+          value: cleaning
+        - dps_val: wall_clean
+          value: cleaning
+        - dps_val: spot_clean
+          value: cleaning
+        - dps_val: mop_clean
+          value: cleaning
+        - dps_val: goto_charge
+          value: returning
+        - dps_val: charging
+          value: charging
+        - dps_val: charge_done
+          value: charged
+        - dps_val: paused
+          value: pause
+        - dps_val: cleaning
+          value: cleaning
+        - dps_val: sleep
+          value: sleep
+    - id: 14
+      type: string
+      name: fan_speed
+      #tuya constants are interpretted backwards by Ultenic
+      #Ultenic App has a 4th "Strong" suction speed but observing
+      #the DP always seems to call it gentle when selected - so dont know what custom logic they are doing
+      mapping:
+        - dps_val: strong
+          value: "Off"
+        - dps_val: normal
+          value: Gentle
+        - dps_val: gentle
+          value: Normal
+
+    - id: 110
+      type: bitfield
+      name: error
+      #have no idea what the fields are for this bit field.  The Robot upside down reports as -2601
+      optional: true
+
+    - id: 111
+      name: Unknown Value (DP 111)
```

Comment:
name of dps is not a human readable name, use "unknown_111".

Reply:
will do

Reply:
done